core/deliverservice/blocksprovider/blocksprovider.go

```diff
@@ -159,14 +162,30 @@ func (b *blocksProviderImpl) DeliverBlocks() {
 
 			marshaledBlock, err := proto.Marshal(t.Block)
 			if err != nil {
-				logger.Errorf("[%s] Error serializing block with sequence number %d, due to %s", b.chainID, blockNum, err)
+				logger.Errorf("[%s] Error serializing block with sequence number %d, due to %s; Disconnecting client from orderer.", b.chainID, blockNum, err)
+				maxDelay := float64(maxRetryDelay)
+				currDelay := float64(time.Duration(math.Pow(2, float64(verErrCounter))) * 100 * time.Millisecond)
+				time.Sleep(time.Duration(math.Min(maxDelay, currDelay)))
+				if currDelay < maxDelay {
+					verErrCounter++
+				}
+				b.client.Disconnect()
 				continue
 			}
 			if err := b.mcs.VerifyBlock(gossipcommon.ChainID(b.chainID), blockNum, marshaledBlock); err != nil {
-				logger.Errorf("[%s] Error verifying block with sequnce number %d, due to %s", b.chainID, blockNum, err)
+				logger.Errorf("[%s] Error verifying block with sequnce number %d, due to %s; Disconnecting client from orderer.", b.chainID, blockNum, err)
```

Comment:
why do we need exponential backoff? this is a problem verifying a block. Waiting more time won't make this block any better.

Reply:
The disconnect is immediately followed by a reconnect. With no backoff, and in case the orderer or orderers are faulty (maybe only on a single channel) there will a constant barrage of connect attempts. It does not necessarily need to be exponential though.